### PR TITLE
[py] python3.7+ syntax throughout `*.py` files.

### PR DESCRIPTION
**Pull request by symonk**
Upgrades all python2 syntax to be python3.7+ (for the cases where `from __future__ import annotations` is defined benefits from more modern typing syntax `|`.  For now this is a full sweep of `/py/*.py`

Related to #10646 

Noticeably:

 - Remove redundant args to `open(...)` (default)
 - Replace string formatting with a more modern approach
 - Reduce explicit args to `super()` for inheritance
 - Removes redundant `__future__` imports
 - Removes redundant unicode prefixing `u'foo'` (I removed most of those in a previous pull request: #10562)
 - Replace `set(...)` with literal `{..., ...}`
 - Replace `IOError` with `OSError` as multiple exceptions are now aliased since `py3.3`
 - more

### CODE REVIEW

File: common/devtools/pdl.py [hunk 1]
```diff
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-from __future__ import print_function
```
Comment (symonk) [inline common/devtools/pdl.py:5]: Todo: Revisit `__future__` imports;

File: py/selenium/webdriver/remote/webelement.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def clear(self) -> None:
         """Clears the text if it's a text entry element."""
         self._execute(Command.CLEAR_ELEMENT)
 
-    def get_property(self, name) -> Union[str, bool, WebElement, dict]:
```
Comment (symonk) [inline py/selenium/webdriver/remote/webelement.py:104]: revert union types `|` these are not `3.7` compliant.

File: py/selenium/webdriver/support/color.py [hunk 3]
```diff
@@ -64,12 +64,12 @@ def from_string(cls, str_: str) -> Color:
         import re
 
         class Matcher:
-            match_obj: Optional[Match[str]]
```
Comment (symonk) [inline py/selenium/webdriver/support/color.py:67]: revert

**Comment by AutomatedTester** [comment]
This seems to be changes to things unrelated to the python bindings, let’s have those in a separate PR

David

**Review by AutomatedTester — APPROVED** [review]

**Review by AutomatedTester — APPROVED** [review]

**Comment by AutomatedTester** [comment]
please fix the tox issue and we can merge

**Comment by symonk** [comment]
> please fix the tox issue and we can merge

was broken from `trunk`, i'll pull in latest

**Pull request merged**